### DISCUSSION TITLE
Image Caption | Rich-text UI Editing

### DISCUSSION THREAD

**Discussion by DevGlitch** [tinacms/tinacms#3794] [Ideas]
It would great to have the possibility to add an actual caption to an image when editing in rich-text. 

Currently the "caption" field in the UI is actually the title field which doesn't produce an actual caption below the image.

[<img src="https://user-images.githubusercontent.com/59583446/230655659-f83f6e3a-4969-4eee-93eb-b0edce5feb10.png" width="75%"/>](image.png)

Suggestion:

- Rename actual "caption" field to "title" to ensure consistency in naming
- Create a new field "caption"

Instead of this:

```
![My Alt Text](/my_image.jpg "My Title")
```

It should produce this:

```
![My Alt Text](/my_image.jpg "My Title")
*My Caption*
```

**Comment by appinteractive** [comment]
Why do you need a third field? I use a custom image component and display alt as the title and caption as an actual caption with a <figure/> html tag 🤔

**Comment by DevGlitch** [comment]
I do the same as you currently. The goal here is to actually ensure that we use markdown emphasis to our advantage when users are using the rich-text editing UI. This way it simply generates the caption below the image without any kind of additional dev needed in users' web template.

Markdown emphasis allow you to add easily a caption below an image and is widely used by the community. So let's ensure we take advantage of that and implement this commonly used syntax.

The idea of a third field is to be able to enter different values for title vs. caption. A toggle button to display or not a caption below the image would work too if we assume title is always the same as caption which might not be true for everyone.


This simple markdown syntax:
```
![My Alt Text](/my_image.jpg "My Title")
*My Caption*
```

Automatically generates this:
```
<p>
    <img src="/my_image.jpg" alt="My Alt Text" title="My Title">
    <em>My Caption</em>
</p>
```

**Reply by appinteractive** [reply]
ahh okay did not know that this gets used that way.

**Comment by jeffsee55** [comment]
> Rename actual "caption" field to "title" to ensure consistency in naming

Good point, we'll do that.

As for the other recommendation I'm not sure how to approach this, the image type is actually an inline element, so this is valid syntax:

```
Hello, world. Here's an image ![My Alt Text](/my_image.jpg "My Title") and some more text.
```
Where would the caption go in that scenario?

https://astexplorer.net/#/gist/516f7bc2ab63cb2e44e2b81787356b9f/4fca2d65f8814a5121ca2f747ab1dca9fd4aeee2

**Reply by DevGlitch** [reply]
> ```
> Hello, world. Here's an image ![My Alt Text](/my_image.jpg "My Title") and some more text.
> ```
> 
> Where would the caption go in that scenario?
> 
> https://astexplorer.net/#/gist/516f7bc2ab63cb2e44e2b81787356b9f/4fca2d65f8814a5121ca2f747ab1dca9fd4aeee2

Good question! This definitely might not be that simple to implement due to that kind of scenario. :/

I guess doing this in our own CSS will have to do for now.
